src/main/java/com/yahoo/identity/services/session/LoggedInSessionImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.yahoo.identity.services.session;
+
+import com.yahoo.identity.services.account.Account;
+import com.yahoo.identity.services.account.AccountService;
+import com.yahoo.identity.services.account.AccountUpdate;
+import com.yahoo.identity.services.credential.Credential;
+import com.yahoo.identity.services.credential.CredentialImpl;
+import com.yahoo.identity.services.credential.CredentialService;
+import com.yahoo.identity.services.credential.CredentialServiceImpl;
+import com.yahoo.identity.services.storage.Storage;
+import com.yahoo.identity.services.storage.sql.SqlAccountService;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+import javax.annotation.Nonnull;
+
+public class LoggedInSessionImpl implements LoggedInSession {
+
+    private CredentialService credentialService;
+    private AccountService accountService;
+    private Credential credential;
+    private boolean verified;
+    private String username;
+    private String password;
+
+    public LoggedInSessionImpl(@Nonnull Storage storage) {
+        this.accountService = new SqlAccountService(storage);
+        this.credentialService = new CredentialServiceImpl();
+        this.credential = new CredentialImpl();
+    }
+
+    @Override
+    @Nonnull
+    public String getUsername() {
+        return this.username;
+    }
+
+    public void setUsername(@Nonnull String username) {
```

Comment:
`LoggedInSession` should be immutable.
You can imagine that if someone is given an instance of `Identity`, ideally, he cannot do any malicious with that instance.
Once a user obtain an instance of `LoggedInSession`, that means he has already been authenticated as the user.
So if we make `LoggedInSessionImpl` mutable, that means the user can pretend himself as others.

Reply:
Sorry for my negligence here. Fixed. `LoggedInSession` now is immutable and I also remove some unused `GET` method. `password` is removed from fields of the class since we just verify it once.